spi_nand_flash/src/nand.c

```diff
@@ -281,16 +296,17 @@ esp_err_t spi_nand_flash_read_sector(spi_nand_flash_device_t *handle, uint8_t *b
 
     xSemaphoreTake(handle->mutex, portMAX_DELAY);
 
-    if (dhara_map_read(&handle->dhara_map, sector_id, buffer, &err)) {
+    if (dhara_map_read(&handle->dhara_map, sector_id, handle->read_buffer, &err)) {
         ret = ESP_ERR_FLASH_BASE + err;
     } else if (err) {
         // This indicates a soft ECC error, we rewrite the sector to recover
-        if (dhara_map_write(&handle->dhara_map, sector_id, buffer, &err)) {
+        if (dhara_map_write(&handle->dhara_map, sector_id, handle->read_buffer, &err)) {
             ret = ESP_ERR_FLASH_BASE + err;
         }
     }
 
     xSemaphoreGive(handle->mutex);
+    memcpy(buffer, handle->read_buffer, handle->page_size);
```

Comment:
Proposal: maybe we could introduce a new API expecting DMA buffer as out-parameter? Something like spi_nand_flash_read_sector_dma(..., caller_provided_dma_buffer, ...) to avoid extra memcpy(). In some future update, not in given PR